sql: add more copy error logging

otan opened the pull request:
Release note (sql change): COPY now logs an error during the insert phase on the SQL_EXEC logging channel.

Informs: https://github.com/cockroachdb/cockroach/issues/90656

Release justification: important logging change (bug fix if you will)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/95038)
<!-- Reviewable:end -->

otan committed:
sql: add more copy error logging

Release note (sql change): COPY now logs an error during the insert
phase on the SQL_EXEC logging channel.

Informs: https://github.com/cockroachdb/cockroach/issues/90656

otan commented:
bors r=rafiss
